AEPIdentity/Sources/Identity.swift

```diff
@@ -62,6 +62,19 @@ import Foundation
             return state?.readyForSyncIdentifiers(event: event, configurationSharedState: configSharedState) ?? false
         } else if event.type == EventType.configuration, event.source == EventSource.requestIdentity {
             return MobileIdentities().areSharedStatesReady(event: event, sharedStateProvider: getSharedState(extensionName:event:))
+        } else if event.type == EventType.identity, event.source == EventSource.requestIdentity, ( event.baseUrl != nil ||  event.urlVariables ) {
+            // check for hub sharedstate info and analytics registration info only if the Analytics shared state is not already set
+            if getSharedState(extensionName: IdentityConstants.SharedStateKeys.ANALYTICS, event: event)?.status != .set {
```

Comment:
From the code for `getSharedState`, it appears if the extension is not registered it will return nil. Can you check with Core team to verify if we can rely on that to determine if an extension is registered or not?
https://github.com/adobe/aepsdk-core-ios/blob/main/AEPCore/Sources/eventhub/EventHub.swift#L230
If that is the case then I think you could change this to something like 
```
if let analyticsSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.Analytics, event: event), analyticsSharedState.status != .set { return false }
```

Reply:
If you cannot simplify this check with the above, can you reduce the number of calls to `getSharedState` to just two? Once for Analytics and once for Event Hub?  Looks like currently there are two calls for Analytics and two for Event Hub.